src/state_machine.zig

```diff
@@ -64,140 +64,74 @@ pub fn StateMachineType(
-            timestamp: u64,
-            ledger: u32,
-            code: u16,
-            flags: AccountFlags,
-            padding: [16]u8,
-
-            comptime {
-                assert(@sizeOf(AccountImmutable) == 64);
-                assert(stdx.no_padding(AccountImmutable));
-            }
-
-            pub fn from_account(a: *const Account) AccountImmutable {
-                return .{
-                    .id = a.id,
-                    .user_data = a.user_data,
-                    .timestamp = a.timestamp,
-                    .ledger = a.ledger,
-                    .code = a.code,
-                    .flags = a.flags,
-                    .padding = mem.zeroes([16]u8),
-                };
-            }
-        };
-
-        pub const AccountMutable = extern struct {
-            debits_pending: u64,
-            debits_posted: u64,
-            credits_pending: u64,
-            credits_posted: u64,
-            timestamp: u64,
-            padding: [24]u8,
-
-            comptime {
-                assert(@sizeOf(AccountMutable) == 64);
-                assert(stdx.no_padding(AccountMutable));
-            }
-
-            pub fn from_account(a: *const Account) AccountMutable {
-                return .{
-                    .debits_pending = a.debits_pending,
-                    .debits_posted = a.debits_posted,
-                    .credits_pending = a.credits_pending,
-                    .credits_posted = a.credits_posted,
-                    .timestamp = a.timestamp,
-                    .padding = mem.zeroes([24]u8),
-                };
-            }
-        };
-
-        pub fn into_account(immut: *const AccountImmutable, mut: *const AccountMutable) Account {
-            assert(immut.timestamp == mut.timestamp);
-            return Account{
-                .id = immut.id,
-                .user_data = immut.user_data,
-                .reserved = mem.zeroes([48]u8),
-                .ledger = immut.ledger,
-                .code = immut.code,
-                .flags = immut.flags,
-                .debits_pending = mut.debits_pending,
-                .debits_posted = mut.debits_posted,
-                .credits_pending = mut.credits_pending,
-                .credits_posted = mut.credits_posted,
-                .timestamp = mut.timestamp,
-            };
-        }
-
-        const AccountsImmutableGroove = GrooveType(
+        const AccountsGroove = GrooveType(
             Storage,
-            AccountImmutable,
+            Account,
             .{
-                .ids = constants.tree_ids.accounts_immutable,
+                .ids = constants.tree_ids.accounts,
                 .value_count_max = .{
                     .timestamp = config.lsm_batch_multiple * @as(usize, @max(
                         constants.batch_max.create_accounts,
                         // ×2 because creating a transfer will update 2 accounts.
                         2 * constants.batch_max.create_transfers,
```

Comment:
`2 * constants.batch_max.create_transfers,` is correct here now that the account is mutable, but previously (i.e. in `main`) where it is immutable, it should just be `constants.batch_max.create_transfers`.

(I'm not posting a fix to main since after this PR it will be correct anyway. :smile:)
Also it shouldn't impact the performance tradeoffs of the `AccountsImmutable`/`AccountMutable` split in either direction.